### PR TITLE
Issue-88098: refactor to avoid import from script

### LINKED ISSUE #88098: [Label Check] Move label check logic from CI to mergebot
Currently, we have [a CI check](https://github.com/pytorch/pytorch/blob/master/.github/workflows/check-labels.yml#L44) that checks to see if a PR has the appropriate labels on it, and fails the check if the right label is missing.

Instead of making this be something that makes the CI appear to have failed, lets make two changes:

1. Modify the CI check to only share a comment reminding the user to add the appropriate CI (without reporting a failing check)
2. Update mergebot rules in https://github.com/pytorch/pytorch/blob/master/.github/scripts/trymerge.py to 1) fail the merge if the PR is missing the required labels and 2) post a comment explaining that the merge was failed due to the missing label

Original customer request:

> Can we make the "needs labels" signal not part of the actual CI signal? I use the GitHub green/red indicator to easily check if a PR succeeded CI or not, but lack of labels turns this into a fail and obscures the signal.
> 
> Instead, **the merge bot should just reject merges if they don't have a label**. The idea being, it's very easy to immediately add the label and try again. You don't need this signal ahead of time!

### PR DESCRIPTION
**Pull request by Goldspear**
Fixes #88098

## Context

During the [initial PR](https://github.com/pytorch/pytorch/pull/92682), we've uncovered some further issues caused by cross script files importing and unnecessary module dependencies. To clean up the repo structure, it's better to extract common fcns / classes to appropriate utils modules, so they can be cleanly shared between various scripts.

## What Changed
* Extracted `GitHubPR` class from `trymerge.py` to `gitutils.py`
* Extracted label checking related fcns into `label_utils.py`

## Note to Reviewers
This is only a 50% review with a draft proposal for how the extraction could work. The test modules will be updated accordingly, once we have a chance to discuss, address feedback and settle on the extraction strategy. Please ignore the failed tests for now.

### CODE REVIEW

**Comment by Goldspear** [comment]
cc @ZainRizvi and @malfet for review and comments

File: .github/scripts/gitutils.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
 #!/usr/bin/env python3
 
+import json
 import os
 import re
 import tempfile
+import github_contants as gh_constants
```
Comment (ZainRizvi) [inline .github/scripts/gitutils.py:7]: typo in 'constants'

File: .github/scripts/github_contants.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
+RE_GHSTACK_HEAD_REF = re.compile(r"^(gh/[^/]+/[0-9]+/)head$")
+RE_GHSTACK_DESC = re.compile(r'Stack.*:\r?\n(\* [^\r\n]+\r?\n)+', re.MULTILINE)
+RE_PULL_REQUEST_RESOLVED = re.compile(
+    r'Pull Request resolved: '
+    r'https://github.com/(?P<owner>[^/]+)/(?P<repo>[^/]+)/pull/(?P<number>[0-9]+)',
+    re.MULTILINE
+)
+RE_PR_CC_LINE = re.compile(r'^cc:? @\w+.*\r?\n?$', re.MULTILINE)
+RE_DIFF_REV = re.compile(r'^Differential Revision:.+?(D[0-9]+)', re.MULTILINE)
```
Comment (ZainRizvi) [inline .github/scripts/github_contants.py:9]: For stuff that's only used in a single file, I'd default to keeping it in that file

File: .github/scripts/gitutils.py (hunk 1 above)
Comment (ZainRizvi) [reply .github/scripts/gitutils.py:7]: Also, this file might be better named "gh_queries"

File: .github/scripts/gitutils.py [hunk 3]
```diff
@@ -277,6 +283,599 @@ def amend_commit_message(self, msg: str) -> None:
...
+class GitHubComment:
+    body_text: str
+    created_at: str
+    author_login: str
+    author_association: str
+    editor_login: Optional[str]
+    database_id: int
+
+
+class GitHubPR:
```
Comment (ZainRizvi) [inline .github/scripts/gitutils.py:483]: Fyi, there are a few other scripts that also reference this class from trymerge.py which will need to be pointed to this one

https://cs.github.com/?q=GitHubPR%20language%3APython%20repo%3Apytorch%2Fpytorch&scopeName=pytorch&scope=org%3Apytorch

File: .github/scripts/gitutils.py [hunk 4]
```diff
@@ -277,6 +297,906 @@ def amend_commit_message(self, msg: str) -> None:
...
+            repo.checkout(orig_branch)
+        return commit
+
+    def delete_land_time_check_branch(self,
+                                      repo: GitRepo) -> None:
+        land_check_branch = f'landchecks/{self.pr_num}'
+        repo._run_git('push', 'origin', '-d', land_check_branch)
+
+
+def read_merge_rules(repo: Optional[GitRepo], org: str, project: str) -> List[MergeRule]:
```
Comment (ZainRizvi) [inline .github/scripts/gitutils.py:972]: This prob wasn't supposed to be here?

**Pull request closed without merging**